### PR TITLE
added wildcard support for entity_ids in 'include'; name is chosen based on match

### PR DESCRIPTION
**Pull request by daringer**
This allows nice tables with the same attribute across many entities, especially fancy for power consumption...
power_consumption

sorry, if my javascript is awful, more a backend guy can for sure update if needed...

### CODE REVIEW

**Comment by daringer** [comment]
might also be good to simply take the ".friendly_name" if not overwritten in the include/attribute list by name---I am not entirely convinced that this looks good enough...

File: entity-attributes-card/entity-attributes-card.js [hunk 1]
```diff
@@ -5,34 +5,58 @@ class EntityAttributesCard extends HTMLElement {
...
           if (filters.every(filterFunc => filterFunc(`${key}.${attr_key}`))) {
+
+	    // based on which part of the URI was machted the (distinguishing) 
+	    // name is chosen as label for this attribute 
+	    // to be fully consistent, shall wildcarding also work for the domain? 
+	    let my_name = '';
+	    if (filter.name) {
+	      my_name = filter.name;
+	    } else {
+	      const res = _getPart(filter);
```
Comment (ciotlosm) [inline entity-attributes-card/entity-attributes-card.js]: You are going backward at this. we execute filterFunc for this exact purpose to do matching.

File: entity-attributes-card/entity-attributes-card.js [hunk 2]
```diff
@@ -5,34 +5,58 @@ class EntityAttributesCard extends HTMLElement {
...
       }
-      const regEx = new RegExp(`^${attribute.replace(/\*/g, '.*')}$`, 'i');
+      const regEx = new RegExp(`^${res.attr.replace(/\*/g, '.*')}$`, 'i');
       return stateObj.search(regEx) === 0;
     }
+
     const attributes = new Map();
     filters.forEach((filter) => {
       const filters = [];
       filters.push(stateObj => _filterName(stateObj, filter));
```
Comment (ciotlosm) [inline entity-attributes-card/entity-attributes-card.js]: the _getPart should be added here not inside an existing filter. Make sure they execute in correct order. I might not understand completely what you're trying to achieve. If you could please include some configuration as well with the picture.

**Comment by daringer** [comment]
thanks for the feedback, will try to refine this during the week.
For now first a config example:
```
- type: custom:entity-attributes-card                                                                                                                                                                                                    
  title: Power Consumption                                                      
  heading_name: Who                                                            
  heading_state: Power (W)                                                     
  filter:                                                                      
    include:                                                                   
      - key: sensor.*_power.power_consumption
```
This shows me all "power_consumption" attributes from all (sensor)entities.

**Comment by daringer** [comment]
Hey hey,

yeees, honestly saying I roughly came to the same conclusion: *abusing* entity-attributes-card for my intentions (showing arbitrary, tabular data) has led to nowhere.
Obviously, I was looking for something different, thus I've come up with **flex-table-card**. Which might be seen as a more generic approach for tabular data, see yourself: https://github.com/daringer/custom-lovelace/tree/master/flex-table-card

So, once I do the final cleanups in the readme and stuff, would you maybe like to include it into your repo? or rather not, then I would simply maintain it in a separate repository.

cheers

**Pull request closed without merging**